### PR TITLE
Support arbitrary types in partitioning columns

### LINKED ISSUE #878: Support non-String partitioning columns
We've encountered non-String partitioning columns in Parquet. We should support those.

### PR DESCRIPTION
**Pull request by rcaudy**
Closes #878

### CODE REVIEW

File: DB/src/main/java/io/deephaven/db/v2/locations/impl/TableLocationFactory.java [hunk 1]
```diff
@@ -23,5 +23,5 @@
      *                       {@link TableLocationProvider#supportsSubscriptions() supports subscriptions}
      * @return A new or cached {@link TableLocation} identified by the supplied {@link TableKey} and {@link TableLocationKey}
      */
-    TableLocation makeLocation(@NotNull final TK tableKey, @NotNull TLK locationKey, @Nullable TableDataRefreshService refreshService);
+    TableLocation makeLocation(@NotNull TK tableKey, @NotNull TLK locationKey, @Nullable TableDataRefreshService refreshService);
```
Comment (jcferretti) [inline DB/src/main/java/io/deephaven/db/v2/locations/impl/TableLocationFactory.java:26]: final dropped but no later change on the file?  guessing you may have been changing the code and needing mutating the argument but then that wasn't the case in the final version?

**Review by jcferretti — APPROVED** [review]

File: DB/src/main/java/io/deephaven/db/v2/locations/impl/TableLocationFactory.java (hunk 1 above)
Comment (rcaudy) [reply DB/src/main/java/io/deephaven/db/v2/locations/impl/TableLocationFactory.java:26]: I was looking at this code for some reason, and realized I had `final` declared on an interface method (which is silly/useless) so got rid of it.

File: DB/src/main/java/io/deephaven/db/v2/sources/regioned/RegionedColumnSourceByte.java [hunk 2]
```diff
@@ -78,14 +80,20 @@ public byte getByte(final long elementIndex) {
     static final class Partitioning extends RegionedColumnSourceByte<Values> {
 
         Partitioning() {
-            super(ColumnRegionByte.createNull(), Supplier::get);
+            super(ColumnRegionByte.createNull(), Supplier::get /* No need to interpose a deferred region in this case. */);
```
Comment (jcferretti) [inline DB/src/main/java/io/deephaven/db/v2/sources/regioned/RegionedColumnSourceByte.java:83]: I find comments inside statements annoying.   Could you consider splitting the argument to a separate line
and adding it as a // comment?

**Review by jcferretti — DISMISSED** [review]

File: DB/src/main/java/io/deephaven/db/v2/sources/regioned/RegionedColumnSourceByte.java (hunk 2 above)
Comment (rcaudy) [reply DB/src/main/java/io/deephaven/db/v2/sources/regioned/RegionedColumnSourceByte.java:83]: Sigh, OK. I thought this was better/clearer.

**Review by jcferretti — APPROVED** [review]

**Pull request merged**